ui/components/app/wallet-overview/eth-overview.js

```diff
@@ -88,15 +95,15 @@ const EthOverview = ({ className }) => {
                 })}
                 data-testid="eth-overview__primary-currency"
                 value={balance}
-                type={PRIMARY}
+                type={treatSecondaryCurrencyAsPrimary ? SECONDARY : PRIMARY}
                 ethNumberOfDecimals={4}
                 hideTitle
               />
               {balanceIsCached ? (
                 <span className="eth-overview__cached-star">*</span>
               ) : null}
             </div>
-            {showFiat && (
+            {showFiat && conversionRate && (
```

Comment:
are there any cases where we should showFiat if conversionRate is null? If not could we move the check for conversionRate into the getShouldShowFiat selector?

Reply:
Good question. Let me poke around!

Reply:
Moved this into a selector and consolidated in a few places as a result. There isn't anywhere I can think/find that we should showFiat if we don't have a valid conversion rate!